File: src/jobsearch.py
import requests
import json

url = 'https://jobsearch.api.jobtechdev.se'
url_for_search = f"{url}/search"

def _get_ads(params):
    headers = {'accept': 'application/json'}
    response = requests.get(url_for_search, headers=headers, params=params)
    response.raise_for_status()  # check for http errors
    return json.loads(response.content.decode('utf8'))


def search_number_of_hits(query):
    # limit: 0 means no ads, just a value of how many ads were found.
    search_params = {'q': query, 'limit': 0}
    json_response = _get_ads(search_params)
    number_of_hits = json_response['total']['value']
    return number_of_hits


def example_search_loop_through_hits(query):
    # limit = 100 is the max number of hits that can be returned.
    # If there are more (which you find with ['total']['value'] in the json response)
    # you have to use offset and multiple requests to get all ads.
    search_params = {'q': query, 'limit': 100}
    json_response = _get_ads(search_params)
    hits = json_response['hits']
    for hit in hits:
        print(f"{hit['headline']}, {hit['employer']['name']}")


def search_jobs(job, location):
    query = f'{job} {location}'
    example_search_loop_through_hits(query)

File: static/js/jobsearch.js
const url = 'https://jobsearch.api.jobtechdev.se';
const urlForSearch = `${url}/search`;


async function getAds(params) {
    const queryString = new URLSearchParams(params).toString();
    const response = await fetch(`${urlForSearch}?${queryString}`, {
        headers: { 'Accept': 'application/json' }
    });
    
    if (!response.ok) {
        throw new Error(`HTTP error! Status: ${response.status}`);
    }

    const data = await response.json();
    return data;
}

async function searchNumberOfHits(query) {
    const searchParams = { q: query, limit: 0 };
    const jsonResponse = await getAds(searchParams);
    const numberOfHits = jsonResponse.total.value;
    return numberOfHits;
}


async function searchAdsEmployer(query) {
    const searchParams = { q: query, limit: 100 };
    const jsonResponse = await getAds(searchParams);
    const hits = jsonResponse.hits;
    return hits.map(hit => {
        return {
            employerName: hit.employer.workplace,
            url: hit.webpage_url,
            location: hit.workplace_address.region,
            // Include other properties you need from the hit object
            // For example:
            // jobTitle: hit.job.title,
            // etc.
        };
    });
}